sources/packages/backend/apps/api/src/route-controllers/announcements/announcement.institutions.controller.ts

```diff
@@ -0,0 +1,30 @@
+import { Controller, Get, Query } from "@nestjs/common";
+import BaseController from "../BaseController";
+import { ApiTags } from "@nestjs/swagger";
+import { Announcement } from "@sims/sims-db";
+import { AnnouncementService } from "../../services";
+import { InstitutionAnnouncementsAPIInDTO } from "../../route-controllers/announcements/models/announcement.dto";
+import { AllowAuthorizedParty } from "../../auth/decorators";
+import { AuthorizedParties } from "../../auth";
+import { ClientTypeBaseRoute } from "../../types";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("announcements")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-announcements`)
+export class AnnouncementInstitutionsController extends BaseController {
+  constructor(private readonly announcementService: AnnouncementService) {
+    super();
+  }
+
+  /**
+   * Get system announcements.
+   * @param target location for the announcement.
+   * @returns system announcements list.
+   */
+  @Get()
+  async getAnnouncements(
+    @Query() queryString: InstitutionAnnouncementsAPIInDTO,
+  ): Promise<Announcement[]> {
+    return this.announcementService.getAnnouncements(queryString.target);
```

Comment:
It is still returning the entity object on API. It must return DTO. I see the OUTDTO being created but not used here. Here is how we can do it.

```ts
  @Get()
  async getAnnouncements(
    @Query() queryString: StudentAnnouncementsAPIInDTO,
  ): Promise<AnnouncementAPIOutDTO[]> {
    const announcements = await this.announcementService.getAnnouncements(
      queryString.target,
    );
    return announcements.map((announcement) => ({
      messageTitle: announcement.messageTitle,
      message: announcement.message,
      startDate: announcement.startDate,
      endDate: announcement.endDate,
      target: announcement.target,
    }));
  }
}
``

Reply:
![image](https://github.com/user-attachments/assets/7d7f9dec-3561-43bf-9248-d32ca0db4279)

Reply:
updated